Src/VimCore/MotionUtil.fs

```diff
@@ -2925,6 +2925,38 @@ type internal MotionUtil
             else
                 x.MarkLine mark
 
+    /// Operate on the next match for last pattern searched for
+    member x.NextMatch path count =
+        let last = _vimData.LastSearchData
+        if StringUtil.IsNullOrEmpty last.Pattern then
+            _statusUtil.OnError Resources.NormalMode_NoPreviousSearch
+            None
+        else
+            let searchKind = SearchKind.OfPathAndWrap path _globalSettings.WrapScan
+            let options = last.Options ||| SearchOptions.IncludeStartPoint
+            let searchData = SearchData(last.Pattern, last.Offset, searchKind, options)
+
+            // All search operations update the jump list.
+            _jumpList.Add x.CaretVirtualPoint
+
+            let searchResult = _search.FindNextPattern x.CaretPoint searchData _wordNavigator count
+
+            // Raise the messages that go with this given result.
+            CommonUtil.RaiseSearchResultMessage _statusUtil searchResult
+
+            let motionResult = 
+                match searchResult with
+                | SearchResult.Error _ -> None
+                | SearchResult.NotFound _ -> None
+                | SearchResult.Found (_, span, _, _) ->
+
+                    let motionKind = MotionKind.CharacterWiseExclusive
+                    let isForward = path = SearchPath.Forward
+                    MotionResult.Create(span, motionKind, isForward) |> Some
+
+            _vimData.ResumeDisplayPattern()

```

Comment:
Why is this needed here? I didn't see anything obvious in this function that would kill the display pattern here.

Reply:
I was unable to leverage `x.SearchCore` because the semantics were so different, but I did use it as a reference as if I were using it.  In this case, `ResumeDisplayPattern` is the second to last line of `x.SearchCore`.

My assumption is that if the user has manually disabled `hlsearch` that any kind of search continuation should reenable it. In other words, `gn` has the same semantics relative to `hlsearch` as `n` does.